pkg/ingester/ingester_test.go

```diff
@@ -3206,10 +3209,64 @@ func TestIngester_OpenExistingTSDBOnStartup(t *testing.T) {
 				require.NotNil(t, i.getTSDB("user2"))
 				require.NotNil(t, i.getTSDB("user3"))
 				require.NotNil(t, i.getTSDB("user4"))
+				walReplayConcurrency := getWALReplayConcurrencyFromTSDBHeadOptions(i.getTSDB("user0"))
+				require.Equal(t, 2, walReplayConcurrency)
 			},
 		},
-		"should fail and rollback if an error occur while loading a TSDB on concurrency > number of TSDBs": {
-			concurrency: 10,
+		"should load all TSDBs on walReplayConcurrency > number of TSDBs": {
+			walReplayConcurrency: 10,
+			setup: func(t *testing.T, dir string) {
+				require.NoError(t, os.MkdirAll(filepath.Join(dir, "user0", "dummy"), 0700))
+				require.NoError(t, os.MkdirAll(filepath.Join(dir, "user1", "dummy"), 0700))
+				require.NoError(t, os.MkdirAll(filepath.Join(dir, "user2", "dummy"), 0700))
+				require.NoError(t, os.MkdirAll(filepath.Join(dir, "user3", "dummy"), 0700))
+				require.NoError(t, os.MkdirAll(filepath.Join(dir, "user4", "dummy"), 0700))
+			},
+			check: func(t *testing.T, i *Ingester) {
+				require.Equal(t, 5, len(i.tsdbs))
+				require.NotNil(t, i.getTSDB("user0"))
+				require.NotNil(t, i.getTSDB("user1"))
+				require.NotNil(t, i.getTSDB("user2"))
+				require.NotNil(t, i.getTSDB("user3"))
+				require.NotNil(t, i.getTSDB("user4"))
+				walReplayConcurrency := getWALReplayConcurrencyFromTSDBHeadOptions(i.getTSDB("user0"))
+				require.Equal(t, 10, walReplayConcurrency)
+			},
+		},
+		"should load all TSDBs on number of TSDBs > maxTSDBOpenWithoutConcurrency": {
+			walReplayConcurrency: 2,
+			setup: func(t *testing.T, dir string) {
+				require.NoError(t, os.MkdirAll(filepath.Join(dir, "user0", "dummy"), 0700))
```

Comment:
[nit] I would simplify both `setup()` and `check()` this way:

```
for _, userID := range []string{"user0", "user1", ..., "user10"} {
    // do stuff for 1 user
}
```

This way, in the `check()`, you can also easily assert on `getWALReplayConcurrencyFromTSDBHeadOptions()` for every user, not just 1 of them.

_Same comment applies to other cases of this test._